pilot/cmd/pilot-agent/status/server.go

```diff
@@ -382,7 +388,8 @@ func (s *Server) handleAppProbe(w http.ResponseWriter, req *http.Request) {
 		if h.Name == "Host" || h.Name == ":authority" {
 			// Probe has specific host header override; honor it
 			appReq.Host = h.Value
-			break
+		} else {
```

Comment:
Aha, i was surprised when seeing this PR since I remembered  this is already handled. See line 380. Then I realized the above is for `req`, sending from the kubelet real request and we deserialize the json to the `prober` only which contains the original header specification.

Could you do a quick validation of what's the header we put after re-writen? I think it's empty and if so, we can delete line 380 blocks.

Reply:
```yaml
    - name: ISTIO_KUBE_APP_PROBERS
      value: '{"/app-health/httpbin/readyz":{"httpGet":{"path":"/get","port":80,"scheme":"HTTP","httpHeaders":[{"name":"foo","value":"bar"}]},"timeoutSeconds":1}}'
...
      failureThreshold: 30
      httpGet:
        path: /healthz/ready
        port: 15021
        scheme: HTTP
      initialDelaySeconds: 1
      periodSeconds: 2
      successThreshold: 1
      timeoutSeconds: 1
```

So alternatively we could just put it in the rewrite. I think this is a bit simpler though, let me delete l380

Reply:
actually l380 is different, its for user agent header. So we should keep both

Reply:
only user agent header is present? and i assume that's set by kubelet? 
keeping both might be safe, though i guess user agent header 95% cases won't make any difference if it's by kubelet.

Reply:
In my experience yes - not 100% sure if thats the only possible setting. Keep in my non-kubelet's could end up calling this endpoint as well directly with whatever headers they want.